core/commonTest/src/kotlinx/serialization/SerializersLookupNamedCompanionTest.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization
+
+import kotlinx.serialization.builtins.*
+import kotlinx.serialization.descriptors.*
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.encoding.*
+import kotlinx.serialization.test.runSince
+import kotlin.reflect.*
+import kotlin.test.*
+
+class SerializersLookupNamedCompanionTest {
+    @Serializable
+    class Plain(val i: Int) {
+        companion object Named
```

Comment:
I said 'add', not 'replace' :) You need at least one test with explicit annotation to verify that all new reflective stuff is working now.